### PR TITLE
Feature/nlp counter

### PR DESCRIPTION
**Pull request by steveoni**
This pull request implements the python `collections.Counter` like, containing the use-case as shown below.

```js
const counter = new Counter({ 'good': 2, 'bad': 2, 'neutral': 3 });
counter.update({'good':4,'bad':2,'food':1})

counter.mostCommon(2) // [ ['good', 6], ['bad', 4] ]

counter.element // [ ['good', 6], ['bad', 4], ['neutral', 3], ['food', 1]]
counter.wordCount // {good': 6, 'bad': 4, 'neutral': 3, 'food':1}
```

### CODE REVIEW

File: src/text/counter.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+type CounterType = CounterObject | string[] | string;
+
+
+export default class Counter {
+  public wordCount: CounterObject = {};
+  public element: Array<Array<string | number>> ;
+
+  /**
+   * Initializer COunter class
```
Comment (yorkie) [inline src/text/counter.ts:14]:
```suggestion
   * Initializer Counter class
```

File: src/text/counter.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+   * create WordCOunt dictionary and array of element
+   * with containing word and their count in order of
+   * increasing decreaing value
+   * e.g
+   * const counter = new Counter(["a","boy","girl"])
+   * counter.wordCount // {'a':1, "boy":1, "girl":1}
+   *
+   * @param textInput
+   */
+  constructor (textInput: CounterType){
```
Comment (yorkie) [inline src/text/counter.ts:24]:
```suggestion
  constructor (textInput: CounterType) {
```

File: src/text/counter.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if (Array.isArray(textInput)){
```
Comment (yorkie) [inline src/text/counter.ts:25]: Ditto

File: src/text/counter.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      this.wordCountFromArray(textInput);
+
+    } else if (typeof textInput === "string"){
+      let toArray = textInput.split('');
```
Comment (yorkie) [inline src/text/counter.ts:29]:
```suggestion
      const toArray = textInput.split("");
```

File: src/text/counter.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+   * const counter = new Counter(["a","boy","girl"])
+   * counter.update({'a':2,'boy':4,'eat':2})
+   * counter.wordCount // {'a': 3, 'boy':5, 'girl':1,'eat':2}
+   *
+   * @param updateInput
+   */
+  public update(updateInput: CounterType): void {
+    const updateCounter = new Counter(updateInput);
+
+    for (const key in updateCounter.wordCount){
```
Comment (yorkie) [inline src/text/counter.ts:52]:
```suggestion
    for (const key in updateCounter.wordCount) {
```

**Review by yorkie — CHANGES_REQUESTED** [review]

**Comment by yorkie** [comment]
@steveoni @risenW Shall we have set up a coding style check, though?

**Comment by steveoni** [comment]
> @steveoni @risenW Shall we have set up a coding style check, though?

Yeah. I'm using Prettier plugins with the Eslint configs, I thought that will handle this style issue. I will check it out and make another commit

**Review by yorkie — APPROVED** [review]

File: test/text/counter.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    
+    assert.deepEqual(counter.wordCount,expectedWordCount);
+    assert.deepEqual(counter.element, expectedElement);
+  });
+  it('should fetch the top 2 common word', ()=>{
+    const counter = new Counter({ 'good': 1, 'bad': 2, 'neural': 3 });
+    const expectedElement = [ [ 'neural', 3 ], [ 'bad', 2 ] ]
+    assert.deepEqual(counter.mostCommon(2), expectedElement);
+  });
+});
```
Comment (FeelyChau) [inline test/text/counter.ts:67]: missing EOF :)

File: src/text/counter.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+   * the number specified.
+   * @param count number
+   * @return Array<Array<string | number>>
+   */
+  public mostCommon(count: number): Array<Array<string | number>> {
+    return this.element.slice(0, count);
+  }
+}
+
+
```
Comment (FeelyChau) [inline src/text/counter.ts:103]: Ditto.

File: src/text/counter.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface CounterObject {
+  [property: string] : number
+}
+
+type CounterType = CounterObject | string[] | string;
+
+
```
Comment (FeelyChau) [inline src/text/counter.ts:8]: Unnecessary blank line.

**Review by FeelyChau — DISMISSED** [review]

**Comment by yorkie** [comment]
@FeelyChau your reviews are style nits from Pipcook, but I think datacook owns itself, it depends on datacook team @risenW @steveoni and @WenheLI :)

**Comment by FeelyChau** [comment]
> @FeelyChau your reviews are style nits from Pipcook, but I think datacook owns itself, it depends on datacook team @risenW @steveoni and @WenheLI :)

Yeah, you are right. Maybe I should enable the lint in another PR if you wish.

**Comment by yorkie** [comment]
@FeelyChau Ok, that's great for the community, and I dismissed yours because they are invalid to this PR :)

**Pull request merged**